src/main/java/com/pikachu/purple/application/user/port/out/UserRepository.java

```diff
@@ -1,7 +1,23 @@
 package com.pikachu.purple.application.user.port.out;
 
+import com.pikachu.purple.domain.user.User;
+import com.pikachu.purple.domain.user.enums.SocialLoginProvider;
+
 public interface UserRepository {
 
+    User findByEmailAndSocialLoginProvider(
+        String email,
+        SocialLoginProvider socialLoginProvider
+    );
+
     void validateNotExistedEmail(String email);
 
+    void save(User user);
+
+    User getById(Long userId);
+
+    void validateNotExistedNickname(String nickname);
+
+    int countTotalUsers();
```

Comment:
p4)
```suggestion
    int countAll()
```
 같은 네이밍이 어떨까요..?